src/resolvers/userResolver.js

```diff
@@ -134,6 +134,62 @@ const userResolver = {
       );
     },
 
+    changeUsername: async(_, { userInfo }, { req }) => {
+      const { newUsername } = userInfo;
+      const oldUser = await User.findById(req.session.userId);
+      const oldUsername = oldUser.username;
+
+      if(oldUsername === newUsername) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "old and new username are same",
+            },
+          ],
+        };
+      }
+
+      let user = await User.findOne({ username: oldUsername });
+      if(!user) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "username does not exists",
+            },
+          ],
+        };
+      }
+
+      try {
```

Comment:
I think we should update the user with [.findByIdAndUpdate()](https://mongoosejs.com/docs/api.html#model_Model-findByIdAndUpdate). That way, we will take advantage of `mongoose` and it'll be much easier to understand what is going on.